Sources/BaseButtonBarPagerTabStripViewController.swift

```diff
@@ -0,0 +1,290 @@
+/*****************************************************************************
+ * BaseButtonBarPageTabStripViewController.swift
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright (c) 2018 VideoLAN. All rights reserved.
+ * $Id$
+ *
+ * Authors: Carola Nitz <caro # videolan.org>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import Foundation
+
+class VLCLabelCell: UICollectionViewCell {
+
+    @IBOutlet weak var iconLabel: UILabel!
+
+}
+public enum SwipeDirection {
+    case left
+    case right
+    case none
+}
+
+public struct IndicatorInfo {
+
+    public var title: String?
+    public var accessibilityLabel: String?
+
+    public init(title: String) {
+        self.title = title
+        self.accessibilityLabel = title
+    }
+}
+
+public enum PagerScroll {
+    case no
+    case yes
+    case scrollOnlyIfOutOfScreen
+}
+
+open class BaseButtonBarPagerTabStripViewController<ButtonBarCellType: UICollectionViewCell>: PagerTabStripViewController, PagerTabStripDataSource, PagerTabStripIsProgressiveDelegate, UICollectionViewDelegate, UICollectionViewDataSource {
+
+    public var changeCurrentIndexProgressive: ((_ oldCell: ButtonBarCellType?, _ newCell: ButtonBarCellType?, _ progressPercentage: CGFloat, _ changeCurrentIndex: Bool, _ animated: Bool) -> Void)?
+
+    public var buttonBarView: ButtonBarView!
+
+    lazy private var cachedCellWidths: [CGFloat]? = { [unowned self] in
+        return self.calculateWidths()
+        }()
+
+    public override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        delegate = self
+        datasource = self
+    }
+
+    @available(*, unavailable, message: "use init(nibName:)")
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let flowLayout = UICollectionViewFlowLayout()
+        flowLayout.scrollDirection = .horizontal
+        buttonBarView = ButtonBarView(frame: .zero, collectionViewLayout: flowLayout)
```

Comment:
You're right, it's not really the laziness that mattered the most to me here!
What I wanted us to consider is a "closure initialization"(not sure about the name 🤔) so that all(or most of) the `buttonBarView` related initialization is done next to its declaration.

One of the big benefits of it would be to have a more "segregated/readable" code by:

- "Containing" the initialization at the top
- Reducing `viewDidLoad` size

Small example in `VLCActionSheet`